File: calls_firebase_cloud_functions/functions/src/notifications/callData.ts
export class CallData {
  readonly callerId: string;
  readonly calleeId: string;
  readonly channelId: string;
  readonly callerPhone: string | undefined;
  readonly callerName: string | undefined;
  readonly hasVideo: boolean | undefined;

  constructor(
    callerId: string,
    calleeId: string,
    channelId: string,
    callerPhone: string | undefined,
    callerName: string | undefined,
    hasVideo: boolean | undefined
  ) {
    this.callerId = callerId;
    this.calleeId = calleeId;
    this.channelId = channelId;
    this.callerPhone = callerPhone;
    this.callerName = callerName;
    this.hasVideo = hasVideo;
  }
}

File: calls_app/android/app/src/main/kotlin/com/tbrgroup/in_app_calls_demo/models/CallData.kt
package com.tbrgroup.in_app_calls_demo.models

import android.os.Parcelable
import kotlinx.parcelize.Parcelize
import org.json.JSONObject

@Parcelize
class CallData(
        val channelId: String?,
        val callerId: String?,
        val callerPhone: String?,
        val callerName: String?,
        val hasVideo: Boolean?,
) : Parcelable {

    constructor(json: JSONObject) : this(
            channelId = if (json.has("channel_id")) json["channel_id"] as String? else null,
            callerId = if (json.has("caller_id")) json["caller_id"] as String? else null,
            callerPhone = if (json.has("caller_phone")) json["caller_phone"] as String? else null,
            callerName = if (json.has("caller_name")) json["caller_name"] as String? else null,
            hasVideo = if (json.has("has_video")) json["has_video"] as Boolean? else null
    )

    fun hasVideoToInt(): Int {
        if (hasVideo == true) return 1
        return 0
    }

    fun toMap(): HashMap<String, Any?> {
        return hashMapOf(
                "channelId" to channelId,
                "callerId" to callerId,
                "callerPhone" to callerPhone,
                "callerName" to callerName,
                "hasVideo" to hasVideo
        )
    }
}